src/Symfony/Bridge/Doctrine/Validator/Constraints/UniqueEntityValidator.php

```diff
@@ -241,4 +276,42 @@ private function formatWithIdentifiers(ObjectManager $em, ClassMetadata $class,
 
         return sprintf('object("%s") identified by (%s)', $idClass, implode(', ', $identifiers));
     }
+
+    public function getFieldValues($object, ClassMetadata $class, array $fields, bool $isEntity = false): array
+    {
+        if (!$isEntity) {
+            $reflectionObject = new \ReflectionObject($object);
+        }
+
+        $fieldValues = [];
+        $objectClass = $object::class;
+
+        foreach ($fields as $objectFieldName => $entityFieldName) {
+            if (!$class->hasField($entityFieldName) && !$class->hasAssociation($entityFieldName)) {
+                throw new ConstraintDefinitionException(sprintf('The field "%s" is not mapped by Doctrine, so it cannot be validated for uniqueness.', $entityFieldName));
+            }
+
+            $fieldName = \is_int($objectFieldName) ? $entityFieldName : $objectFieldName;
+            if (!$isEntity) {
+                if (!$reflectionObject->hasProperty($fieldName)) {
+                    throw new ConstraintDefinitionException(sprintf('The field "%s" is not a property of class "%s".', $fieldName, $objectClass));
+                }
+            }
+
+            $fieldValues[$entityFieldName] = $this->getPropertyValue($objectClass, $fieldName, $object);
+        }
+
+        return $fieldValues;
+    }
+
+    public function getPropertyValue(string $class, string $name, $object): mixed
+    {
+        $property = new \ReflectionProperty($class, $name);
+
+        if (!$property->isInitialized($object)) {
+            return null;
+        }
```

Comment:
Should we let this throw an exception? It feels like a misconfiguration by the user. Or, perhaps we throw our own clear exception if this happens to help them fix the misconfig.

Reply:
I have already mentioned it here https://github.com/symfony/symfony/pull/38662#discussion_r759530585
In your opinion, what kind of exception should this be?

Reply:
I see! I don't have a big preference. Maybe we just remove the check and let php throw an exception. And if people are eventually confused by this, we an add a better exception later.